src/OrleansSQLUtils/Storage/AdoNetConfigurationExtensions.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Orleans.Runtime.Configuration;
+using Orleans.Storage;
+using System;
+using System.Collections.Generic;
+
+
+namespace OrleansSQLUtils
+{
+    /// <summary>
+    /// The format in which the data will serialized to the storage.
+    /// </summary>
+    public enum AdoNetSerializationFormat
+    {
+        /// <summary>
+        /// The native Orleans binary serialization format.
+        /// </summary>
+        OrleansNativeBinary = 1,
+
+        /// <summary>
+        /// XML.
+        /// </summary>
+        Xml                 = 2,
+
+        /// <summary>
+        /// JSON.
+        /// </summary>
+        Json                = 3
+    }
+
+
+    /// <summary>
+    /// Extension methods for configuration classes specific to <see cref="OrleansSQLUtils"/>.
+    /// </summary>
+    public static class AdoNetConfigurationExtensions
+    {
+        /// <summary>
+        /// Adds a storage provider of type <see cref="AdoNetStorageProvider"/>.
+        /// </summary>
+        /// <param name="config">The cluster configuration object to add provider to.</param>
+        /// <param name="providerName">The provider name.</param>
+        /// <param name="connectionString">The ADO.NET storage connection string. If none is provided, it will use the same as in the Globals configuration.</param>
+        /// <param name="serializationFormat">The format in which the data will serialized to the storage.</param>
+        public static void AddAdoNetStorageProvider(
+            this ClusterConfiguration config,
+            string providerName = "AdoNetStorage",
+            string connectionString = null,
+            AdoNetSerializationFormat serializationFormat = AdoNetSerializationFormat.Xml)
+        {
+            if(string.IsNullOrWhiteSpace(providerName))
+            {
+                throw new ArgumentNullException(nameof(providerName));
+            }
+
+            connectionString = GetConnectionString(connectionString, config);
```

Comment:
I understand that could be an issue in native code, but the reality is that not here in c# when we are dealing with references to immutable objects/structs.
But to add to it, IMO creating a new variable just so that we don't update the reference can cause more maintainability problems than just reusing it (as code could mistakenly use the input parameter instead of the sanitized one). Also for consistency with everywhere else we should let this in and if you feel strongly (I don't, so I'm open to change) we can open an issue and then update all our usages of input sanitization in one go, but not be prescriptive yet on the non-consistent approach until there's consensus.